joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/astcreation/AstCreator.scala

```diff
@@ -129,7 +129,7 @@ class AstCreator(
   protected def column(node: Node): Option[Integer]    = node.getBegin.map(x => Integer.valueOf(x.column)).toScala
   protected def lineEnd(node: Node): Option[Integer]   = node.getEnd.map(x => Integer.valueOf(x.line)).toScala
   protected def columnEnd(node: Node): Option[Integer] = node.getEnd.map(x => Integer.valueOf(x.column)).toScala
-  protected def code(node: Node): String               = "" // TODO: javasrc2cpg uses custom code strings everywhere
+  protected def code(node: Node): String               = node.clone.removeComment().toString
```

Comment:
Just out of curiosity: is the clone call actually necessary?

Reply:
Indeed, I don't like it either. Fortunately, it's not "recursive" and we are only using it in this patch for basic nodes - javasrc2cpg crafts the `code` itself for more complex nodes.

`removeComment` destructively updates the node, and I didn't want to bring any side-effects into this, hence the clone. Other than that, I first tried to do `getTokenRange`, filter out comment nodes (cf. `getCategory`), but that has the disadvantage of missing whitespace - e.g. `1 + 2` would be written as `1+2`, since whitespace tokens are not in the token range. Design-wise, comments are attached to nodes, but I couldn't find a reasonable way to discard them other than with `removeComment` -- maybe @johannescoetzee knows, as he's more familiar with javaparser than I am?

Reply:
Alternatively, this removal of comment nodes is mostly relevant when we're dealing with literals, since their code is their value. I'm happy to restrict this patch to those only until we find a better solution, as well.

Reply:
`removeComment` modifies the original node, so it is necessary except if we specifically handle comments before code (although javasrc2cpg doesn't currently handle comments at all, if I recall correctly)

Reply:
> I'm happy to restrict this patch to those only until we find a better solution

In my opinion, handling that case specifically adds more code complexity which outweighs the performance gain. With that being said, it does look like `clone` does do a deep clone, so an uglier, but more performant option that doesn't rely on handling any specific case could be to remove the comment, get the string representation and then add the comment again